From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: qeth: remove unnecessary support ckeck in sysfs route6
References: bnc#434333

Symptom:     route6 attribute can not be set in initial device condition
Problem:     Usability
Solution:    Remove this check improves usability because you do not have to
             set the device online to initially set ipv6 routing option.

Signed-off-by: Frank Blaschka <frank.blaschka@de.ibm.com>

Acked-by: John Jolly <jjolly@suse.de>
---

 drivers/s390/net/qeth_l3_sys.c |    7 -------
 1 file changed, 7 deletions(-)

Index: linux-sles11/drivers/s390/net/qeth_l3_sys.c
===================================================================
--- linux-sles11.orig/drivers/s390/net/qeth_l3_sys.c
+++ linux-sles11/drivers/s390/net/qeth_l3_sys.c
@@ -121,9 +121,6 @@ static ssize_t qeth_l3_dev_route6_show(s
 	if (!card)
 		return -EINVAL;
 
-	if (!qeth_is_supported(card, IPA_IPV6))
-		return sprintf(buf, "%s\n", "n/a");
-
 	return qeth_l3_dev_route_show(card, &card->options.route6, buf);
 }
 
@@ -135,10 +132,6 @@ static ssize_t qeth_l3_dev_route6_store(
 	if (!card)
 		return -EINVAL;
 
-	if (!qeth_is_supported(card, IPA_IPV6)) {
-		return -EOPNOTSUPP;
-	}
-
 	return qeth_l3_dev_route_store(card, &card->options.route6,
 				QETH_PROT_IPV6, buf, count);
 }
